src/modules/job-manager/plugins/dependency-after.c

```diff
@@ -0,0 +1,563 @@
+                                       scheme,
+                                       jobid);
+
+    if (!(after = after_info_create (id, type, jobid)))
+        return flux_jobtap_reject_job (p, args,
+                                       "failed to establish job dependency");
+
+    /*  Emit the dependency
+     */
+    if (flux_jobtap_dependency_add (p, id, after->description) < 0) {
+        after_info_destroy (after);
+        return flux_jobtap_reject_job (p, args, "Unable to add job dependency");
+    }
+
+    /*  Corner case, requisite job may have already started. Check for that
+     *   here and immediately satisfy dependency before adding to various
+     *   lists below
+     */
+    if (type == AFTER_START
+        && (target_state == FLUX_JOB_STATE_RUN
+            || target_state == FLUX_JOB_STATE_CLEANUP
+            || target_state == FLUX_JOB_STATE_INACTIVE)) {
+        if (flux_jobtap_dependency_remove (p, id, after->description) < 0)
+            flux_log_error (flux_jobtap_get_flux (p),
+                            "flux_jobtap_dependency_remove");
+        after_info_destroy (after);
+        return 0;
+    }
+
+    /*  Append this dependency to the deplist in the target jobid:
+     */
+    if (!(l = after_list_get (p, afterid))
+        || !(handle = zlistx_add_end (l, after))) {
+        after_info_destroy (after);
+        return flux_jobtap_reject_job (p,
+                                       args,
+                                       "failed to append to list");
+    }
+
+    /*  Create a reference in the current job to the depednency, so it can
+     *   be removed if this job terminates before PRIORITY state.
+     */
+    if (!(ref = after_ref_create (afterid, l, after))
+        || !(l = after_refs_get (p, id))
+        || !zlistx_add_end (l, ref)) {
+        after_info_destroy (after);
+        after_ref_destroy (ref);
+        return flux_jobtap_reject_job (p, args, "failed to create ref");
+    }
+
+    return 0;
+}
+
+
+/*  Attempt to remove the job dependency described by `after`. If this
+ *   fails, try to raise a fatal job exception on the current job.
+ */
+static void remove_jobid_dependency (flux_plugin_t *p,
+                                     struct after_info *after)
+{
+    if (flux_jobtap_dependency_remove (p,
+                                       after->depid,
+                                       after->description) < 0) {
+        if (flux_jobtap_raise_exception (p,
+                                         after->depid,
+                                         "dependency",
+                                         0,
+                                         "Failed to remove dependency %s",
+                                         after->description) == 0) {
+            flux_log_error (flux_jobtap_get_flux (p),
+                            "flux_jobtap_raise_exception: id=%ju",
+                            (uintmax_t) after->depid);
+        }
```

Comment:
Use of `flux_log_error()` when errno not defined?

Reply:
errno should be set by `flux_jobtap_raise_exception()`, is there some path where you see it is not?

Reply:
Oh maybe the test for error is wrong then. `flux_jobtap_raise_exception()` returns -1 on error.

Reply:
Oh, strange, I didn't even see that! I don't know why that `== 0` is there. Thanks for catching it.